src/start_server.js

```diff
@@ -41,36 +52,48 @@ function startWithPort(options) {
 
   console.log('Starting Polyserve on port ' + options.port);
 
-  var app = express();
-  var polyserve = makeApp(options.componentDir, options.packageName);
+  let app = express();
+  let polyserve = makeApp({
+    componentDir: options.componentDir,
+    packageName: options.packageName,
+    root: process.cwd(),
+  });
 
   app.use('/components/', polyserve);
 
-  var server = http.createServer(app);
+  let server = http.createServer(app);
+  let serverStartedResolve;
+  let serverStartedReject;
+  let serverStartedPromise = new Promise((resolve, reject) => {
+    serverStartedResolve = resolve;
+    serverStartedReject = reject;
+  });
 
-  server = app.listen(options.port, options.host);
+  server = app.listen(options.port, options.host,
+      () => serverStartedResolve(server));
 
   server.on('error', function(err) {
-    if (err.code === 'EADDRINUSE')
-      console.error("ERROR: Port in use", options.port,
-        "\nPlease choose another port, or let an unused port be chosen automatically.");
-    process.exit(69);
+    if (err.code === 'EADDRINUSE') {
+      console.error(portInUseMessage(options.port));
+    }
+    serverStartedReject(err);
   });
 
-  var baseUrl = util.format('http://'+ options.host +':%d/components/%s/', options.port,
-    polyserve.packageName);
-  console.log('Files in this directory are available under ' + baseUrl);
+  let baseUrl = `http://${options.host}:${options.port}/components/${polyserve.packageName}/`;
+  console.log(`Files in this directory are available under ${baseUrl}`);
 
   if (options.page) {
-    var url = baseUrl + (options.page === true ? 'index.html' : options.page);
+    let url = baseUrl + (options.page === true ? 'index.html' : options.page);
     if (Array.isArray(options.browser)) {
-      for (var i = 0; i < options.browser.length; i++)
+      for (let i = 0; i < options.browser.length; i++)
         opn(url, options.browser[i]);
     }
     else {
       opn(url, options.browser);
     }
   }
```

Comment:
Yeah, but here we're calling app.listen and then immediately doing a console.log() and starting browsers, right?

I'm suggesting that we should put that logic inside of `serverStartedPromise.then(() => { ... })`